anvio/panops.py

```diff
@@ -89,12 +90,18 @@ def __init__(self, args=None, run=run, progress=progress):
         self.enforce_hierarchical_clustering = A('enforce_hierarchical_clustering')
         self.enforce_the_analysis_of_excessive_number_of_genomes = anvio.USER_KNOWS_IT_IS_NOT_A_GOOD_IDEA
 
+        self.de_novo_compute_mode = A('mode') or 'sequence'
```

Comment:
somewhere in `check_params` function we should check for the use of `--pan-mode structure` together with `--user-defined-gene-clusters` parameter (and prevent it basically).